File: structr-ui/src/main/resources/structr/js/structr.js
/* 
 *  Copyright (C) 2013 Axel Morgner
 * 
 *  This file is part of structr <http://structr.org>.
 * 
 *  structr is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU Affero General Public License as
 *  published by the Free Software Foundation, either version 3 of the
 *  License, or (at your option) any later version.
 * 
 *  structr is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 * 
 *  You should have received a copy of the GNU Affero General Public License
 *  along with structr.  If not, see <http://www.gnu.org/licenses/>.
 */
var debug = true;
var wsRoot = '/structr/ws';
var reconn, loggedIn;
var token = '';
var rawResultCount = [], pageCount = [];

function connect() {

    if (token) {
        loggedIn = true;
    }

    try {

        ws = null;

        var isEnc = (window.location.protocol === 'https:');
        var host = document.location.host;
        var wsUrl = 'ws' + (isEnc ? 's' : '') + '://' + host + wsRoot + '?' + location.pathname;

        console.log(wsUrl);
        if ('WebSocket' in window) {

            ws = new WebSocket(wsUrl, 'structr');

        } else if ('MozWebSocket' in window) {

            ws = new MozWebSocket(wsUrl, 'structr');

        } else {

            alert('Your browser doesn\'t support WebSocket.');
            return false;

        }

        //console.log('WebSocket.readyState: ' + ws.readyState, ws);

        ws.onopen = function() {

            //console.log('de-activating reconnect loop', reconn);
            window.clearInterval(reconn);
        }

        ws.onclose = function() {

            if (reconn) {
                console.log('Automatic reconnect already active');
                return;
            }
            console.log('activating reconnect loop');
            reconn = window.setInterval(function() {
                connect();
            }, 1000);
            console.log('activated reconnect loop', reconn);

        }

        ws.onmessage = function(message) {

            var data = JSON.parse(message.data);
            console.log('ws.onmessage:', data);
            var command = data.command;

            //console.log('####################################### ', command, ' #########################################');

            if (command === 'PARTIAL') {

                console.log(data.message);
                var pos = data.data.parentPositionPath.split('/');
                var el = document.childNodes[1]; // start with html
                pos.forEach(function(p) {
                  var n = parseInt(p);
                  if (n) {
                    el = getChildElements(el)[n];
                    console.log(el);
                  }
                });
                el.innerHTML = data.message;
            }
        }

    } catch (exception) {
        console.log('Error in connect(): ' + exception);
    }

}

function getChildElements(el) {
  if (!el) return;
  var children = [];
  for (var i=0; i<el.childNodes.length; i++) {
    var c = el.childNodes[i];
    if (c.nodeType === 1) children.push(c);
  }
  return children;
}

connect();

File: structr-db-driver-api/src/main/java/org/structr/api/config/Structr.java
/**
 * Copyright (C) 2010-2016 Structr GmbH
 *
 * This file is part of Structr <http://structr.org>.
 *
 * Structr is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * Structr is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Structr.  If not, see <http://www.gnu.org/licenses/>.
 */
package org.structr.api.config;

/**
 *
 */
public interface Structr {

	public static final String DATABASE_DRIVER               = "database.driver";
	public static final String DATABASE_DRIVER_MODE          = "database.driver.mode";
	public static final String DATABASE_CONNECTION_URL       = "database.connection.url";
	public static final String DATABASE_CONNECTION_USERNAME  = "database.connection.username";
	public static final String DATABASE_CONNECTION_PASSWORD  = "database.connection.password";

	public static final String DATABASE_PATH                 = "database.path";
	public static final String RELATIONSHIP_CACHE_SIZE       = "database.cache.relationship.size";
	public static final String NODE_CACHE_SIZE               = "database.cache.node.size";
	public static final String QUERY_CACHE_SIZE              = "database.cache.query.size";

	public static final String LOG_CYPHER_DEBUG              = "log.cypher.debug";

	public static final String DEFAULT_DATABASE_URL          = "bolt://localhost:7688";
	public static final String TEST_DATABASE_URL             = "bolt://localhost:7689";
}
